phoebe/atmospheres/models.py

```diff
@@ -0,0 +1,506 @@
+    atmospheres.
+
+    In order to have a model atmosphere supported, the following attributes
+    need to be defined:
+
+    * `name` (string): name of the model atmosphere
+    * `prefix` (string): prefix for the model atmosphere fits keywords
+    * `basic_axis_names` (list): names of the basic axes; basic axes are
+        axes that span the basic n-dimensional model atmosphere grid. The grid
+        can be sparsely populated, but it must be regular.
+    * `mus` (array): specific angles, mu=cos(theta), where theta is the angle
+        between the observer and the surface normal.
+    * `wls` (array): wavelengths of the model atmosphere intensities.
+    * `units` (float): intensity unit conversion factor. The intensities are
+        usually given in erg/s/cm^2/A, which is converted to W/m^3 by
+        multiplying with this factor.
+
+    In addition, the following methods need to or may be overloaded:
+
+    * `parse_rules`: provides rules for parsing atmosphere fits filenames to
+        extract basic axis values.
+    * `limb_treatment`: defines how intensities at the exact limb (mu=0) should
+        be treated. By default, the intensities are linearly extrapolated to
+        mu=0.
+
+    When a model atmosphere is instantiated via the from_path() method, the basic
+    axes are populated with unique values from the filenames of the atmosphere
+    fits files. The axes are then exported as numpy arrays.
+
+    Attributes that are automatically populated:
+
+    * `basic_axes` (tuple): tuple of numpy arrays for basic axes
+
+    In the case of from_path() instantiation, the following attributes are
+    also populated:
+
+    * `models` (list): list of atmosphere fits files
+    * `nmodels` (int): number of atmosphere fits files
+    * `indices` (array): array of indices for all defined nodes in the model
+        atmosphere
+    * `[axis_name]` (array): numpy array for each axis, where the name is
+        automatically inferred from the basic axis names
+
+    Arguments
+    ----------
+    * `basic_axes` (tuple of ndarrays): values of the basic axes
+    * `from_path` (bool): if True, the class is instantiated from a path
+
+    Raises
+    -------
+    * `FileNotFoundError`: if the path does not exist
+    """
+
+    name = None
+    prefix = None
+    external = False
+
+    # default axes:
+    basic_axis_names = ['teffs', 'loggs', 'abuns']
+
+    def __init__(self, basic_axes=None, from_path=False):
+        if from_path:
+            return
+
+        self.path = None
+
+        # the model needs to either provide tabulated intensities or
+        # a function to compute them. If the model provides tabulated
+        # intensities, basic_axes must be defined; if it provides a
+        # function, basic_axes will be automatically determined from
+        # axis definitions via class attributes.
+
+        if hasattr(self, 'intensity') and callable(self.intensity):
```

Comment:
this _might_ be cheaper (I'm actually not sure - take it or leave it 🤷‍♂️ )

```suggestion
        if callable(getattr(self, 'intensity', None)):
```